src/classes/CRLP_Operation_SVC.cls

```diff
@@ -358,7 +358,15 @@ public class CRLP_Operation_SVC {
                         rollup.maxByYear.put(theYear, amountValue);
                     }
                 }
-            } else {
+            } else if (rollup.countByYear != null && rollup.countByYear.containsKey(theYear)) {
```

Comment:
Since it seems the issue is that counter was overwritten, then the new block line 361-367 is not required and current line 370 `rollup.countByYear.put(theYear, 1);` can be replaced with 
```
                if (rollup.countByYear.containsKey(theYear)) {
                    rollup.countByYear.put(theYear, rollup.countByYear.get(theYear) + 1);
                } else {
                    rollup.countByYear.put(theYear, 1);
                }
```

Reply:
Great suggestion @RadSFDeveloper

Reply:
@RadSFDeveloper , I believe it has to be an else if here and not if here. If this is another if condition, then the loop might execute in sum if as well as countbyyear, then the count calculations might not be correct.

Reply:
@RadSFDeveloper , I got it now, line 370 is under the else loop, will change, no issues.